internal/iam/repository_role.go

```diff
@@ -443,8 +448,9 @@ func (r *Repository) queryRoles(ctx context.Context, whereClause string, args []
 		}
 		transactionTimestamp, err = rd.Now(ctx)
 		return err
-	}); err != nil {
-		return nil, time.Time{}, err
+	})
+	if err != nil {
+		return nil, time.Time{}, errors.Wrap(ctx, err, op, errors.WithMsg("failed to query roles"))
```

Comment:
This `errors.WithMsg` is the same as the one inside the transaction. I think it's safe to just return `err` here?